wwwapp/models.py

```diff
@@ -281,3 +305,49 @@ def result_in_percent(self):
 
     class Meta:
         unique_together = [('workshop', 'participant')]
+
+
+class ResourceYearPermission(models.Model):
+    """
+    Resource associated with a WWW edition (year). Resource can be accessed by
+    users who qualified or had a lecture on year. User is granted access to
+    root_url and recursively to all files and subdirectories inside.
+    """
+    display_name = models.CharField(max_length=50, blank=True)
+    access_url = models.URLField(blank=True,
+                                 help_text="URL dla przycisku w menu. Przycisk nie jest wyświetlany jeśli url jest pusty")
+    root_uri = models.CharField(max_length=256, null=False, blank=False,
+                                help_text='bez "/" na końcu i na początku. np. "internety/www15"')
```

Comment:
I would prefer this to always start with a `/` so that it looks like a proper URI

Reply:
I will rename to "path". I checked and URI seems to be a set of strings containg all url-s.

Reply:
The path part of the URI should still start with a `/` :P

```
For the URI http://www.example.org:56789/a/b/c.txt?t=win&s=chess#para5 these would be:

Scheme: http
Authority: www.example.org:56789
  User Information: not present
  Host: www.example.org
  Port: 56789
Path: /a/b/c.txt
Query: t=win&s=chess
Fragment: para5
```